AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/DefaultEncryptedKeyStore.dfy

```diff
@@ -0,0 +1,720 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+include "../Model/AwsCryptographyKeyStoreTypes.dfy"
+include "Structure.dfy"
+include "ErrorMessages.dfy"
+
+module DefaultEncryptedKeyStore {
+  import opened Wrappers
+  import ComAmazonawsDynamodbTypes
+  import Seq
+  import Types = AwsCryptographyKeyStoreTypes
+  import DDB = ComAmazonawsDynamodbTypes
+  import UTF8
+  import Structure
+
+  const ToAttributeMap := Structure.ToAttributeMap
+  const ToEncryptedHierarchicalKey := Structure.ToEncryptedHierarchicalKey
+  import ErrorMessages = KeyStoreErrorMessages
+
+  const BRANCH_KEY_EXISTS_EXPRESSION_ATTRIBUTE_NAME := "#BranchKeyIdentifierField"
+  const BRANCH_KEY_EXISTS_EXPRESSION_ATTRIBUTE_NAMES
+    := map[
+         BRANCH_KEY_EXISTS_EXPRESSION_ATTRIBUTE_NAME := Structure.BRANCH_KEY_IDENTIFIER_FIELD
+       ]
+  const BRANCH_KEY_NOT_EXIST_CONDITION := "attribute_not_exists(" + BRANCH_KEY_EXISTS_EXPRESSION_ATTRIBUTE_NAME + ")"
+  const BRANCH_KEY_EXISTS_CONDITION := "attribute_exists(" + BRANCH_KEY_EXISTS_EXPRESSION_ATTRIBUTE_NAME + ")"
+
+  datatype ConditionExpression =
+    | BRANCH_KEY_NOT_EXIST
+    | BRANCH_KEY_EXISTS
+
+  class {:termination false} DynamoDBEncryptedKeyStore
```

Comment:
Issue: We should distinguish b/w Key Store and Key Storage.

Key Store: Read and Authenticate. Generate+ & Write.

Storage: Read & Write.
```suggestion
  class {:termination false} DynamoDBEncryptedKeyStorage
```